gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -232,6 +238,9 @@ public int getByteLength() {
   /** Identifies this version of the GoogleHadoopFileSystemBase library. */
   public static final String GHFS_ID;
 
+  /** Instrumentation to track Statistics */
+  private GhfsInstrumentation instrumentation;
```

Comment:
I think we could use the delegate approach here as well, also please move the entry point calls to the beginning of the method